src/main/java/com/google/devtools/build/lib/rules/cpp/CcLibrary.java

```diff
@@ -312,10 +312,11 @@ public static void init(
         || !ccCompilationOutputs.isEmpty()) {
       if (featureConfiguration.isEnabled(CppRuleClasses.TARGETS_WINDOWS)) {
         // If user specifies a custom DEF file, then we use it.
-        Artifact defFile = common.getWinDefFile();
-
-        Artifact defParser = common.getDefParser();
+        Artifact customDefFile = common.getWinDefFile();
         Artifact generatedDefFile = null;
+        Artifact trivialDefFile = CppHelper.createTrivialDefFileAction(ruleContext);
```

Comment:
Can we spare creating `trivialDefFile` here too, and create it only in line 346?

Reply:
Done.